runners/flink/src/main/java/org/apache/beam/runners/flink/translation/wrappers/streaming/state/FlinkStateInternals.java

```diff
@@ -419,7 +429,130 @@ public int hashCode() {
     }
   }
 
-  private static class FlinkBagState<K, T> implements BagState<T> {
+  private static class FlinkOrderedListState<T> implements OrderedListState<T> {
+    private final StateNamespace namespace;
+    // TODO: Consider instead storing MapStateDescriptor<Instant, T> - at least for some state
+    // backends this will
+    // be stored sorted, which could save time sorting.
```

Comment:
Do we want to keep this todo? TBH I'm still not big fan of this approach, as it relies on side effects that can change at any time without notice.